components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/io/BundleClassLoaderProvider.java

```diff
@@ -67,6 +69,16 @@ public ClassLoader getClassLoader(String componentName, String componentVersion,
         return bundleWiring.getClassLoader();
     }
 
+    @Override
+    public void deployAPI(Object serviceImplementation, Dictionary<String, ?> properties) {
+        if (!(serviceImplementation instanceof Microservice)) {
+            throw new UUFException("API class " + serviceImplementation.getClass().getName() +
```

Comment:
Let's change the exception message to "Cannot deploy REST API {`serviceImplementation.getClass().getName()`} as a micro-service because it does not implement {`Microservice.class.getName()`}".